pkgs/tools/virtualization/awsebcli/default.nix

```diff
@@ -2,70 +2,52 @@
 let
 
   localPython = python.override {
-    packageOverrides = self: super: rec {
-      cement = super.cement.overridePythonAttrs (oldAttrs: rec {
-        version = "2.8.2";
-
-        src = super.fetchPypi {
-          inherit (oldAttrs) pname;
-          inherit version;
+    packageOverrides = self: super: {
+      cement = super.cement.overridePythonAttrs (oldAttrs: {
+        src = oldAttrs.src.override {
+          version = "2.8.2";
           sha256 = "1li2whjzfhbpg6fjb6r1r92fb3967p1xv6hqs3j787865h2ysrc7";
         };
       });
 
-      colorama = super.colorama.overridePythonAttrs (oldAttrs: rec {
-        version = "0.3.7";
-
-        src = super.fetchPypi {
-          inherit (oldAttrs) pname;
-          inherit version;
+      colorama = super.colorama.overridePythonAttrs (oldAttrs: {
+        src = oldAttrs.src.override {
+          version = "0.3.7";
           sha256 = "0avqkn6362v7k2kg3afb35g4sfdvixjgy890clip4q174p9whhz0";
         };
       });
 
-      docker = super.docker.overridePythonAttrs (oldAttrs: rec {
+      docker = super.docker.overridePythonAttrs (oldAttrs: {
+        src = super.fetchPypi {
           pname = "docker-py";
           version = "1.7.2";
-          name = "${pname}-${version}";
```

Comment:
You can only drop that if you remove `name` from the `pythonPackages.docker` expression.

Reply:
However, it seems to me like the docker dependency and some others were completely removed from awsebcli's setup.py. Please have a look.

Reply:
You're right that `docker-py` is no longer required.  `docker` is now an optional dependency.  Even though it's an optional dependency the script is a bit evil, it will try and install `docker` using `pip` if you run a command that requires it.  I'm not sure how to resolve this though, because the `docker` in `nixpkgs` requires a version of `requests` that `awsebcli` is not satisified with.  I'll remove it for now.

Reply:
Okay, then we should try to add it nevertheless. Is there maybe a patch to make `awsebcli` work with a newer version of `docker-py`?